include/boost/iterator/iterator_facade.hpp

```diff
@@ -368,6 +368,23 @@ namespace iterators {
             return *m_iter;
         }
 
+        // conversion operator to value_type
+        // it is needed in case reference is not convertible to value_type by a standard conversion sequence
+        // (e.g. boost::multi_array iterators)
+        template<typename T = value_type>
```

Comment:
This is C++11, Boost.Iterator must be compatible with C++03.

Reply:
Thanks for pointing that out - I'll revoke this PR anyway as microsoft/stl also revoked their change that initially broke multi_array usage with std::sort (https://github.com/microsoft/STL/pull/289)